Code does not build with clang-20.1.3

Replicate:
1. checkout main (`01c9e6f490033f13322e46235aaa412c2b451027`)
2. build as usual
3. observe:

```
opensn/modules/linear_boltzmann_solvers/discrete_ordinates_problem/sweep/communicators/aah_async_comm.h:44:34: error: no template named 'result_of_t' in namespace 'std'; did you mean 'boost::result_of'?
   44 |            std::is_convertible_v<std::result_of_t<GetUnknownCountFunc(std::size_t)>, std::size_t>
      |                                  ^~~~~~~~~~~~~~~~
      |                                  boost::result_of
```

On [cppreference](https://en.cppreference.com/w/cpp/types/result_of.html), we can see that `result_of` was removed in C++20, so this needs to be updated for the language standard, since we are setting it up to be C++20 in the cmake. The page suggests that `invoke_result_t` should be to C++20 equivalent...

--- a/modules/linear_boltzmann_solvers/discrete_ordinates_problem/sweep/communicators/aah_async_comm.h
+++ b/modules/linear_boltzmann_solvers/discrete_ordinates_problem/sweep/communicators/aah_async_comm.h
@@ -41,7 +41,8 @@ struct AAH_MessageDetails
  */
 template <typename GetUnknownCountFunc>
   requires std::is_invocable_v<GetUnknownCountFunc, std::size_t> &&
-           std::is_convertible_v<std::result_of_t<GetUnknownCountFunc(std::size_t)>, std::size_t>
+           std::is_convertible_v<std::invoke_result_t<GetUnknownCountFunc, std::size_t>,
+                                 std::size_t>
 void
 SetupMessageData(const std::vector<int>& locations,
                  const GetUnknownCountFunc& get_unknown_count,

